app/models/user.rb

```diff
@@ -5,14 +5,15 @@ class User < ActiveRecord::Base
 
   PERMITTED_ATTRS = [:bio, :email, :handle, :hide_email, :location, :password, :website].freeze
 
+  before_destroy :yank_gems
```

Comment:
I just figured it would be safer doing a hard deletion of the gems, something we have learned is not a good idea over the years.

We could keep the gems around - but then they are orphaned if the user being deleted is the only author. At least, that's what a quick pass at the `Pusher#authorize` code looks like:

https://github.com/rubygems/rubygems.org/blob/master/app/models/pusher.rb#L23-L24
https://github.com/rubygems/rubygems.org/blob/master/app/models/rubygem.rb#L180

Pusher can authorize a user to push a given gem if it's an unsaved record, has no indexed versions, or it's owned by the user. We don't have a case yet for no ownerships on the gem.